Utils/auto_update_docker/extract_affected_content_items.py

```diff
@@ -0,0 +1,447 @@
+        affected_docker_images (list[str]): Affected docker images.
+        content_items_by_docker_image (dict[str, list[dict[str, Any]]]): A dictionary that holds docker images as keys,
+        and the value will be a list containing data about the content items and respective pack.
+        images_tag (dict[str, str]): A dictionary of docker images and their tags supplied to the program.
+        nightly_packs (list[str]): The nightly packs.
+
+    Returns:
+        dict[str, dict[str, Any]]: A dictionary where the keys are docker images, and their values are data containing
+        the affected content items, pr tags, and target tag of each docker image.
+    """
+    affected_content_items_by_docker_image: dict[str, Any] = {}
+    custom_images: list[str] = list(image_custom_configs.keys())
+    for docker_image in affected_docker_images:
+        logging.info(f"Getting content items for {docker_image = }")
+        affected_content_items: dict[str, Any] = {}
+        docker_image_target_tag = get_docker_image_target_tag(docker_image=docker_image, images_tag=images_tag)
+        docker_batch_config = get_docker_batch_config(
+            docker_image=docker_image,
+            custom_images=custom_images,
+            default_batches=default_batches,
+            image_custom_configs=image_custom_configs,
+            current_batch_index=current_batch_index,
+        )
+        if docker_batch_config:
+            affected_content_items = filter_content_items_to_run_on(
+                batch_config=docker_batch_config,
+                content_items_coverage=content_items_coverage,
+                content_items_by_docker_image=content_items_by_docker_image[docker_image],
+                target_docker_tag=docker_image_target_tag,
+                nightly_packs=nightly_packs,
+                docker_image=docker_image,
+                benchmark_docker_tags=benchmark_docker_tags
+            )
+        else:
+            logging.info(f"No batch config was found for {docker_image = }, and {current_batch_index = }")
+
+        if not affected_content_items["content_items"]:
+            logging.info(f"{docker_image = } does not have any content items to update")
+        affected_content_items_by_docker_image[docker_image] = affected_content_items
+    return affected_content_items_by_docker_image
+
+
+def calculate_affected_docker_images(
+    docker_images_arg: str,
+    images_to_exclude: list[str],
+    all_docker_images: list[str],
+) -> list[str]:
+    """Calculates the docker images that will be used in the current batch.
+    Docker images in the 'images_to_exclude' list will ALWAYS be excluded.
+    Args:
+        docker_images_arg (str): Docker images arg supplied by the user. This will either be:
+            i) ALL - Use all docker images\n
+            ii) docker1,docker2,... - A list of docker images to use.\n
+            iii) ALL@docker1,docker2,... - All docker images, excluding docker1,docker2,...\n
```

Comment:
why this syntax? is this used elsewhere? ALL@docker1,docker

Reply:
We thought about using the format **ALL/docker1,docker2**, and split using the character **/**, but since docker images themselves can hold the character **/**, the splitting would break, so we needed a character that is not common in docker images.
We even tried **-**, but docker images can have it.